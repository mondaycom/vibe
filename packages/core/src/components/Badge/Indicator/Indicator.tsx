import React from "react";
import cx from "classnames";
import styles from "./Indicator.module.scss";
import { IndicatorColor as IndicatorColorEnum } from "./IndicatorConstants";
import { IndicatorColor } from "./Indicator.types";
import { ComponentDefaultTestId } from "../../../tests/constants";
import { VibeComponentProps } from "../../../types";
import { getTestId } from "../../../tests/test-ids-utils";

export interface IndicatorProps extends VibeComponentProps {
  /**
   * The color of the indicator.
   */
  color?: IndicatorColor;
}

const Indicator: React.FC<IndicatorProps> & { colors?: typeof IndicatorColorEnum } = ({
  color = "notification",
  className,
  id,
  "data-testid": dataTestId
}: IndicatorProps) => {
  return (
    <div
      className={cx(styles.indicator, styles[color], className)}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.INDICATOR, id)}
    />
  );
};

Indicator.colors = IndicatorColorEnum;

export default Indicator;
