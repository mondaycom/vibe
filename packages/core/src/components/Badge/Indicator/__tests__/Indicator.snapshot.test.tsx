import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import Indicator from "../Indicator";
import { IndicatorColor } from "../IndicatorConstants";

describe("Indicator", () => {
  it("renders correctly with empty props", () => {
    const tree = renderer.create(<Indicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a custom color", () => {
    const tree = renderer.create(<Indicator color={IndicatorColor.PRIMARY} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
