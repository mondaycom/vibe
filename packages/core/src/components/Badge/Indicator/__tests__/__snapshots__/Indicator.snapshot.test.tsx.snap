// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Indicator > renders correctly with a custom color 1`] = `
<div
  className="indicator primary"
  data-testid="indicator"
/>
`;

exports[`Indicator > renders correctly with empty props 1`] = `
<div
  className="indicator notification"
  data-testid="indicator"
/>
`;
